applications: Added overlays allowing to support more BT conns. (#12443)
nrfconnect/sdk-nrf: applications/matter_bridge/README.rst

* Temperature Sensor
* Humidity Sensor

All the listed Matter device types are enabled by default.

> To disable one of them, set the following configuration options:

Reviewer: ```suggestion
To disable one of them, set any of the following configuration options:
```

> * :kconfig:option:`CONFIG_BRIDGE_ONOFF_LIGHT_BRIDGED_DEVICE` to ``n``
> * :kconfig:option:`CONFIG_BRIDGE_TEMPERATURE_SENSOR_BRIDGED_DEVICE` to ``n``
> * :kconfig:option:`CONFIG_BRIDGE_HUMIDITY_SENSOR_BRIDGED_DEVICE` to ``n``

Reviewer: ```suggestion
* :kconfig:option:`CONFIG_BRIDGE_ONOFF_LIGHT_BRIDGED_DEVICE` to ``n`` to disable On/Off Light.
* :kconfig:option:`CONFIG_BRIDGE_TEMPERATURE_SENSOR_BRIDGED_DEVICE` to ``n`` to disable Temperature Sensor.
* :kconfig:option:`CONFIG_BRIDGE_HUMIDITY_SENSOR_BRIDGED_DEVICE` to ``n`` to disable Humidity Sensor.
```
While it is obvious from the config name, it's better to be explicit.

* :kconfig:option:`CONFIG_BRIDGED_DEVICE_SIMULATED` - For the simulated bridged device.
* :kconfig:option:`CONFIG_BRIDGED_DEVICE_BT` - For the Bluetooth LE bridged device.

Additionally, you can decide how many bridged devices the bridge application will support.
The decision will make an impact on the flash and RAM memories usage, what can be verified on the compilation level.

> The application uses dynamic memory allocation and stores bridged devices objects on heap, so it may be necessary to increase the heap size using ::kconfig_option::`CONFIG_CHIP_MALLOC_SYS_HEAP_SIZE`.

Reviewer: ```suggestion
The application uses dynamic memory allocation and stores bridged device objects on the heap, so it may be necessary to increase the heap size using the :kconfig_option:`CONFIG_CHIP_MALLOC_SYS_HEAP_SIZE` Kconfig option.
```

Use the following configuration options to customize the number of supported bridged devices:

> * ::kconfig_option::`CONFIG_BRIDGE_MAX_BRIDGED_DEVICES_NUMBER` - For changing the maximum number of non-Matter bridged devices supported by the bridge application
> * ::kconfig_option::`CONFIG_BRIDGE_MAX_DYNAMIC_ENDPOINTS_NUMBER` - For changing the maximum number of Matter endpoints used for bridging devices by the bridge application.
>   This option does not have to be equal to ::kconfig_option::`CONFIG_BRIDGE_MAX_BRIDGED_DEVICES_NUMBER`, as it is possible to use non-Matter devices that are represented using more than one Matter endpoints.

Reviewer: ```suggestion
* :kconfig_option:`CONFIG_BRIDGE_MAX_BRIDGED_DEVICES_NUMBER` - For changing the maximum number of non-Matter bridged devices supported by the bridge application
* :kconfig_option:`CONFIG_BRIDGE_MAX_DYNAMIC_ENDPOINTS_NUMBER` - For changing the maximum number of Matter endpoints used for bridging devices by the bridge application.
  This option does not have to be equal to :kconfig_option:`CONFIG_BRIDGE_MAX_BRIDGED_DEVICES_NUMBER`, as it is possible to use non-Matter devices that are represented using more than one Matter endpoint.
```

Configuring the number of Bluetooth LE bridged devices
------------------------------------------------------

> For the Bluetooth LE bridged devices, you have to consider more factors before deciding about the number of supported bridged devices.

Reviewer: ```suggestion
You have to consider several factors to decide how many Bluetooth LE bridged devices to support.
```

> Every Bluetooth LE bridged devices uses a separate Bluetooth LE connection, so you should increase the number of devices in a consistent way with the number of supported Bluetooth LE connections selected using ::kconfig_option::`CONFIG_BT_MAX_CONN`.

Reviewer: ```suggestion
Every Bluetooth LE bridged device uses a separate Bluetooth LE connection, so you need to consider the number of supported Bluetooth LE connections (selected using the :kconfig:option:`CONFIG_BT_MAX_CONN` Kconfig option) when you configure the number of devices.
```

> Additionally, the Matter stack uses one Bluetooth LE connection for the commissioning, so at most it is possible to use ::kconfig_option::`CONFIG_BT_MAX_CONN` - 1 for the bridged devices purpose.

Reviewer: ```suggestion
Since the Matter stack uses one Bluetooth LE connection for commissioning, the maximum number of connections you can use for bridged devices is one less than is configured using the :kconfig:option:`CONFIG_BT_MAX_CONN` Kconfig option.
```

> Additionally, the Matter stack uses one Bluetooth LE connection for the commissioning, so at most it is possible to use ::kconfig_option::`CONFIG_BT_MAX_CONN` - 1 for the bridged devices purpose.

Author: @wiba-nordic slightly rephrased this sentence. Is it more clear to you now, or you still think it should be clarified more?

> Increasing the number of Bluetooth LE connections affects the RAM usage on both application and network cores.

Reviewer: ```suggestion
Increasing the number of Bluetooth LE connections affects the RAM usage on both the application and network cores.
```

> The current maximum number of Bluetooth LE connections that can be selected using default configuration is ``10``.

Reviewer: ```suggestion
The current maximum number of Bluetooth LE connections that can be selected using the default configuration is ``10``.
```

> You can increase the number of Bluetooth LE connections after decreasing the size of Bluetooth LE TX/RX buffers used by the Bluetooth controller, but keep in mind it will decrease the communication throughput.

Reviewer: ```suggestion
You can increase the number of Bluetooth LE connections if you decrease the size of the Bluetooth LE TX/RX buffers used by the Bluetooth controller, but this will decrease the communication throughput.
```

> To enable configuration that increases the number of Bluetooth LE connections to ``20`` by decreasing the sizes of Bluetooth LE TX/RX buffers, build the target using the following command in the project directory with *absolute_path* replaced by absolute path to the Matter bridge application on your local disk:

Reviewer 2: Hmm, is that expected? Btw. could `$PWD` work?

Author: I'm not quite sure. The fact is that our build system uses hci rpmsg Zephyr sample location, as a starting point for the hci_rpmsg overlay. I don't remember whether it's regression or it worked like this always. 

Regarding $PWD, yes, it will work for Linux. But it won't for Windows.

> To enable configuration that increases the number of Bluetooth LE connections to ``20`` by decreasing the sizes of Bluetooth LE TX/RX buffers, build the target using the following command in the project directory with *absolute_path* replaced by absolute path to the Matter bridge application on your local disk:

Reviewer: ```suggestion
Build the target using the following command in the project directory to enable a configuration that increases the number of Bluetooth LE connections to ``20`` by decreasing the sizes of Bluetooth LE TX/RX buffers:
```
Then add this after the command:
```
Replace *absolute_path* with the absolute path to the Matter bridge application on your local disk.
```

* :kconfig:option:`CONFIG_BRIDGED_DEVICE_SIMULATED` - For the simulated bridged device.
* :kconfig:option:`CONFIG_BRIDGED_DEVICE_BT` - For the Bluetooth LE bridged device.

Additionally, you can decide how many bridged devices the bridge application will support.

> The decision will make an impact on the flash and RAM memories usage, what can be verified on the compilation level.

Reviewer: ```suggestion
The decision will make an impact on the flash and RAM memory usage, and is verified in the compilation stage.
```
